### DISCUSSION TITLE
Example for Vanilla JS app?

### DISCUSSION THREAD

**Discussion by kresnasatya** [TanStack/router#464] [Q&A] [answered]
Hi, is this library cover for vanilla JS app (without framework)? If yes, I would like to try. But, for a moment I see the docs currently give examples for React.

**Comment by tannerlinsley** [comment, marked as answer]
No vanilla yet. It will probably be the last priority we tackle unfortunately. However, if you peek at the React adapter, it might give you an idea of how to wire it up on your own.

**Reply by kresnasatya** [reply]
Thanks for the information, Tanner.

**Reply by oskarrough** [reply]
Where is the react adapter in the source? Would be great if you could use whichever HTML-producing library you prefer.

**Reply by pawelblaszczyk5** [reply]
Afaik, the priorities shifter a bit since then and the library is intended to be React only. So there is no adapter currently, and the code just incorporated React parts into core. It's probably not really doable to make it work with Vanilla

**Reply by oskarrough** [reply]
Ah, shame (for non-react people) but fair enough. Thanks for the heads up!

**Comment by nicoandreoli2000** [comment]
Hi @tannerlinsley @kresnasatya

Here's my implementation of the agnostic routing library I made based on tanstack router source code: https://www.npmjs.com/package/typescript-routes.

It's adaptable to any JS framework + any routing library. I suggest to checkout the example with React + React Router, it really worked for me.

Looking forward to any feedback/comments :)
